packages/shared/client/ReplayClient.ts

```diff
@@ -981,12 +981,10 @@ export class ReplayClient implements ReplayClientInterface {
         const loadedRegions = this.loadedRegions;
         let isLoaded = false;
         if (loadedRegions !== null) {
-          if (focusRange !== null) {
-            isLoaded = isRangeInRegions(focusRange, loadedRegions.loaded);
+          if (focusRange === null) {
+            isLoaded = areRangesEqual(loadedRegions.indexed, loadedRegions.loading);
           } else {
-            isLoaded =
-              loadedRegions.loaded.length > 0 &&
```

Comment:
I removed the `length > 0` check, because AFAIK `loading` can never be an empty array (so if `loaded` is empty they can never be equal), but again, I don't feel super strongly about it. The important part of the change is `loaded` -> `indexed`.

Reply:
I'd prefer to keep that check because we shouldn't rely on undocumented details of the protocol:
- since they're undocumented anyone will feel free to change them in the backend and then the frontend may break
- it should be possible to fully understand the frontend code without insider knowledge that even most replay employees (including myself) don't have

Reply:
I agree with Holger.

Reply:
Alright, fair enough.

Reply:
I've slightly adjusted this logic. It sounds like the correct interpretation is "if we are not currently trying to load anything, then we cannot possibly be loaded yet". If I'm interpreting that correctly, I feel like the easiest way to signal that is to put it in the top level conditional, rather than `&&`ing it together in one of the branches but not the other!